src/dart_diagnostic_provider.ts

```diff
@@ -48,4 +51,12 @@ export class DartDiagnosticProvider {
 				throw new Error("Unknown severity type: " + severity); 
 		}
 	}
+
+	private flushResults(notification: as.AnalysisFlushResultsNotification) {		
+		if (notification.files) {
+			notification.files.forEach(file => {
+				this.diagnostics.delete(Uri.file(file));
```

Comment:
I think it might be better to use the [`set(entries: [Uri, Diagnostic[]][])`](https://code.visualstudio.com/docs/extensionAPI/vscode-api#DiagnosticCollection) method here (and just `.map` on files to convert). This way Code gets all the updates in one go and might avoid updating the list multiple times (though admittedly, it's probably unlikely we'd get a huge number of files in this event).